iac-extensions/kubernetes/src/test/java/org/sonar/iac/helm/LineNumberCommentInserterTest.java

```diff
@@ -126,6 +126,17 @@ void shouldAddCommentLineNumberOnAlreadyCommentedLine() {
     checkHelmProcessing("line1 # some comment", "line1 # some comment #1");
   }
 
+  @Test
+  void shouldNotAddLineCommentWhenSeparateDocument() {
+    checkHelmProcessing(
+      code("line1",
+        "---",
+        "line3"),
+      code("line1 #1",
+        "---",
```

Comment:
Just curious, does snakeyaml generally not support comments after `---` or only at the very end of the file, when the new document consists only of comment? Because I believe that comment after triple dash is allowed in YAML.

Reply:
Yes, the comment is allowed after a triple dash or dot. But snakeyaml-engine doesn't support it :-(